mlrun/datastore/targets.py

```diff
@@ -977,9 +977,14 @@ def as_df(
         **kwargs,
     ):
         df = super().as_df(
-            columns=columns, df_module=df_module, entities=entities, **kwargs
+            columns=columns,
+            df_module=df_module,
+            entities=entities,
+            format="csv",
+            **kwargs,
         )
-        df.set_index(keys=entities, inplace=True)
+        if entities:
```

Comment:
Worth to check it also
```suggestion
        if all(ent in df.columns for ent in entities):
```

Reply:
The suggested code would break, because `entities` is `None`.

Do we have any code that ever passes this parameter? Alternatively, is it documented anywhere?

Reply:
```suggestion
        if entities and all(ent in df.columns for ent in entities):
```

Reply:
Ok, I know, but I asked the questions that I asked to understand if it even makes sense. I can't find any hint for what `entities` would even be set to.